assisted_deployment.sh

```diff
@@ -7,18 +7,181 @@ source common.sh
+ASSISTED_OPERATOR_INDEX="${ASSISTED_OPERATOR_INDEX:-quay.io/ocpmetal/assisted-service-index:latest}"
+
+
+function deploy_local_storage() {
+  oc adm new-project openshift-local-storage || true
+
+  oc annotate project openshift-local-storage openshift.io/node-selector=''
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: local-operator-group
+  namespace: openshift-local-storage
+spec:
+  targetNamespaces:
+    - openshift-local-storage
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: local-storage-operator
+  namespace: openshift-local-storage
+spec:
+  channel: "4.7"
+  installPlanApproval: Automatic
+  name: local-storage-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+EOF
+
+  wait_for_crd "localvolumes.local.storage.openshift.io"
+
+  echo "Creating local volume and storage class..."
+  cat <<EOCR | oc apply -f -
+apiVersion: local.storage.openshift.io/v1
+kind: LocalVolume
+metadata:
+  name: assisted-service
+  namespace: openshift-local-storage
+spec:
+  logLevel: Normal
+  managementState: Managed
+  storageClassDevices:
+    - devicePaths:
+        - /dev/sdb
+        - /dev/sdc
+      storageClassName: assisted-service
+      volumeMode: Filesystem
+EOCR
+}
+
+
+function deploy_hive() {
+  echo "Installing Hive..."
+
+  cat <<EOCR | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: hive-operator
+  namespace: openshift-operators
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: hive-operator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: hive-operator.v1.0.19
+EOCR
+
+  wait_for_crd "clusterdeployments.hive.openshift.io"
+}
+
+
+function deploy_assisted_operator() {
+  echo "Installing assisted-installer operator..."
+
+  cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $ASSISTED_INSTALLER_NAMESPACE
+  labels:
+    name: $ASSISTED_INSTALLER_NAMESPACE
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: assisted-service-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: $ASSISTED_OPERATOR_INDEX
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+    name: assisted-installer-group
+    namespace: $ASSISTED_INSTALLER_NAMESPACE
+spec:
+  targetNamespaces:
+    - $ASSISTED_INSTALLER_NAMESPACE
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: assisted-service-operator
+  namespace: $ASSISTED_INSTALLER_NAMESPACE
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: assisted-service-operator
+  source: assisted-service-catalog
+  sourceNamespace: openshift-marketplace
+  startingCSV: assisted-service-operator.v0.0.1
```

Comment:
For dev-scripts sake, we might as well make all the configuration possible...possible.

```suggestion
  startingCSV: assisted-service-operator.v0.0.1
    spec:
    channel: alpha
    config:
      env:
      - name: OPENSHIFT_VERSIONS
        value: ${OPENSHIFT_VERSIONS}
      - name: SERVICE_IMAGE
        value: ${ASSISTED_INSTALLER_SERVICE_IMAGE}
      - name: DATABASE_IMAGE
        value: ${ASSISTED_INSTALLER_DB_IMAGE}
      - name: AGENT_IMAGE
        value: ${ASSISTED_INSTALLER_AGENT_IMAGE}
      - name: CONTROLLER_IMAGE
        value: ${ASSISTED_INSTALLER_CONTROLLER_IMAGE}
      - name: INSTALLER_IMAGE
        value: ${ASSISTED_INSTALLER_IMAGE}
```